packages/ubuntu_desktop_installer/linux/my_application.cc

```diff
@@ -11,6 +11,31 @@ struct _MyApplication {
 
 G_DEFINE_TYPE(MyApplication, my_application, GTK_TYPE_APPLICATION)
 
+static void my_application_get_workarea(GtkWindow* window,
+                                        GdkRectangle* workarea) {
+  GdkWindow* gdk_window = gtk_widget_get_window(GTK_WIDGET(window));
+  GdkDisplay* gdk_display = gdk_window_get_display(gdk_window);
+  GdkMonitor* monitor =
+      gdk_display_get_monitor_at_window(gdk_display, gdk_window);
+  gdk_monitor_get_workarea(monitor, workarea);
+}
+
+static gboolean my_application_fit_to_workarea(GtkWindow* window) {
+  GtkAllocation allocation;
+  gtk_widget_get_allocation(GTK_WIDGET(window), &allocation);
+
+  GdkRectangle workarea;
+  my_application_get_workarea(window, &workarea);
+
+  gboolean fits_workarea =
+      allocation.width < workarea.width || allocation.height < workarea.height;
+  if (!fits_workarea) {
+    gtk_window_move(window, workarea.x, workarea.y);
+    gtk_window_resize(window, workarea.width, workarea.height);
```

Comment:
Perhaps instead of explicitly moving and resizing the window could be maximized instead? That way the window manager is responsible for correct placement.

Reply:
Thanks! Much better :)